gsplat/cuda/csrc/world_to_cam_bwd.cu

```diff
@@ -83,7 +90,7 @@ world_to_cam_bwd_kernel(const uint32_t C, const uint32_t N,
             for (uint32_t i = 0; i < 3; i++) { // rows
                 PRAGMA_UNROLL
                 for (uint32_t j = 0; j < 3; j++) { // cols
-                    gpuAtomicAdd(v_covars + i * 3 + j, v_covar[j][i]);
+                    gpuAtomicAdd(v_covars + i * 3 + j, T(v_covar[j][i]));
                 }
```

Comment:
How should we make sure we are not missing all the necessary casting? What kind of tests do we need to add?

Reply:
I'm not 100% sure the explicit cast in all cases. The compiler complains in the atomicAdd case  since the type of `v_covars` is `T*` and `v_covar[j][i]` is `OpT` and there's no matching function definition for `gpuAtomicAdd`. So for all atomic ops, the compiler will catch these for us. 

Otherwise, remark that in general all the outputs are global writes of the form 
```c++
template <typename T>
__global__ kernel(..., T* kernel_output) {
    OpT computed_value = ...;
 
    ...
    
    kernel_output_var[threadIdx] = T(computed_value); 
}
```  

If you don't have the explicit cast on the last line, then the cast is implicitly done, which is what we want anyway. You may get a warning though which you can fix and make the code more readable.